File: req/mcmc1.py
import numpy as np
import random as rand

rounds = []
nsims = 10000000
p = 0.5
for i in range(1, nsims):
    r = 0
    nloss = 0
    while nloss != 2:
        r+=1
        if rand.random() < p:
            nloss = 0
        else: 
            nloss += 1
    rounds.append(r)

print(np.average(rounds))

File: req/mcmc1.cpp
#include <iostream>
#include <time.h>
using  namespace std;

int main()
{
    int nsims = 100000;
    int rounds [nsims];
    float p = 0.5;
    for (int i = 0; i <= nsims; i++)    {
        
        int r = 0;
        int nloss = 0;
        
        while (nloss != 2)  {
        
            r+=1;
            
            if ((float) rand()/RAND_MAX < p){ nloss = 0;}

            else{ nloss += 1;}
        rounds[i] = r;
        
        }
    }

    double m = 0;
    for (int i = 0; i <= nsims; i++)    {
        
        m  = m + rounds[i];
    
    }
    m = m / (float) nsims;

    cout << m <<endl<<"6";
    return 0;
}
